src/Symfony/Component/Serializer/Tests/Normalizer/AbstractNormalizerTest.php

```diff
@@ -191,8 +193,7 @@ public function testObjectWithVariadicConstructorTypedArguments(AbstractNormaliz
         $d2->qux = 'QUZ';
         $obj = new VariadicConstructorTypedArgsDummy($d1, $d2);
 
-        $serializer = new Serializer([$normalizer], [new JsonEncoder()]);
-        $normalizer->setSerializer($serializer);
+        $serializer = new Serializer([], [new JsonEncoder()], new ChainNormalizer([$normalizer]), new ChainDenormalizer([$normalizer]));
```

Comment:
```suggestion
        $serializer = new Serializer([], [new JsonEncoder()], $normalizer, $normalizer);
```
Same for other occurrences